### PR TITLE
Format changes for using OAuth

### PR DESCRIPTION
**Pull request by royeom**
This has to change for merging. 

Many **OAuth** files are added.

Please take a look.

### CODE REVIEW

File: backend/__init__.py [hunk 1]
```diff
@@ -1,21 +0,0 @@
-from flask import Flask
-from flask import request
-from flask import Blueprint
-from flask_restful import Api
-
-from restful.register import Register
-from restful.login import Login
-
-
```
Comment (harrydrippin) [inline backend/__init__.py:9]: Remove this line.

File: backend/quadcore/restful/github/login_github.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from flask import Flask, redirect, url_for, session, request, jsonify
+from flask_restful import Resource, Api
+from quadcore.manager.auth import AuthManager as am
+from quadcore import app, oauth
+
+class LoginGithub(Resource):
+
+    def get(self):
+        return am.oauth_github.authorize(callback="http://api.quadcore.news/login/github/authorized", _external=True)
```
Comment (harrydrippin) [inline backend/quadcore/restful/github/login_github.py:9]: Provide `next` parameter for proper user flow.

File: backend/quadcore/restful/github/login_github.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        # return {
+        #     "result": 1,
+        #     "reason": "You approached wrong way!"
+        # }
+
+    def post(self):
+        return am.oauth_github.authorize(callback="http://api.quadcore.news/login/github/authorized", _external=True)
```
Comment (harrydrippin) [inline backend/quadcore/restful/github/login_github.py:16]: Remove this line.

File: backend/quadcore/restful/github/logout_github.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+    def get(self):
+        return {
+            "result": 1,
+            "reason": "You approached wrong way!"
+        }
+
+    def post(self):
+        session.pop('github_token', None)
+        return 'OK'
```
Comment (harrydrippin) [inline backend/quadcore/restful/github/logout_github.py:15]: Returning string is not recommended. Try to package data to Python dictionary, which will reformat to JSON.

**Review by harrydrippin — COMMENTED** [review]
Thanks!

File: backend/quadcore/manager/auth.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+from quadcore import oauth
+
+class AuthManager:
+
+    oauth_github = oauth.remote_app(
+        'github',
+        consumer_key='Iv1.2ce83a98d79af095',
+        consumer_secret='05316b10734d10fa53790d109f8ddc7484e8758e',
```
Comment (ethkim) [inline backend/quadcore/manager/auth.py:8]: api keys shouldn't be checked into repository - Note that this repository is public, so anyone can use this.

File: backend/quadcore/manager/auth.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        access_token_url='https://github.com/login/oauth/access_token',
+        authorize_url='https://github.com/login/oauth/authorize',
+        redirect_uri='http://quadcore.news/profile'
+    )
+
+    @oauth_github.tokengetter
+    def get_github_oauth_token():
+        return session.get('github_token')
+
+    # def change_linkedin_query(uri, headers, body):
```
Comment (ethkim) [inline backend/quadcore/manager/auth.py:22]: Please delete all commented out code - repository should be free of any left over code. (You could keep that in your branch)

File: backend/quadcore/restful/github/login_github.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+from flask import Flask, redirect, url_for, session, request, jsonify
+from flask_restful import Resource, Api
+from quadcore.manager.auth import AuthManager as am
+from quadcore import app, oauth
+
+class LoginGithub(Resource):
+
+    def get(self):
+        return am.oauth_github.authorize(callback="http://api.quadcore.news/login/github/authorized", _external=True)
+        # return {
```
Comment (ethkim) [inline backend/quadcore/restful/github/login_github.py:10]: Remove commented code.

**Review by ethkim — COMMENTED** [review]
Thanks!

File: backend/.gitignore [hunk 8]
```diff
@@ -0,0 +1,3 @@
+__pycache__
+.DS_Store
+*.pyc
```
Comment (shberry) [inline backend/.gitignore]: Should there be a new line at the end of the file?

File: backend/quadcore/manager/auth.py [hunk 9]
```diff
@@ -0,0 +1,38 @@
+from quadcore import oauth
+
+class AuthManager:
+
+    oauth_github = oauth.remote_app(
+        'github',
+        consumer_key='Iv1.2ce83a98d79af095',
```
Comment (shberry) [inline backend/quadcore/manager/auth.py]: Is it safe to hardcode the key here? Same below

**Pull request closed without merging**